lua/lspconfig/vdmj.lua

```diff
@@ -0,0 +1,127 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+if vim.fn.has 'nvim-0.5.1' == 0 then
+  vim.notify('The VDMJ language server requires nvim > 0.5', vim.log.levels.ERROR)
+  return
+end
+
+local server_name = 'vdmj'
+
+local mavenrepo = util.path.join(vim.env.HOME, '/.m2/repository/com/fujitsu')
+
+local function get_jar_path(config, package, version)
+  return util.path.join(config.options.mavenrepo, package, version, package .. '-' .. version .. '.jar')
+end
+
+local function with_precision(version, is_high_precision)
+  return is_high_precision and version:gsub('([%d.]+)', '%1-P') or version
+end
+
+local function get_latest_installed_version(repo)
+  local path = util.path.join(repo, 'lsp')
+  local sort = vim.fn.sort
+
+  local subdirs = function(file)
+    local stat = vim.loop.fs_stat(util.path.join(path, file))
+    return stat.type == 'directory' and 1 or 0
+  end
+
+  local candidates = vim.fn.readdir(path, subdirs)
+  local sorted = sort(sort(candidates, 'l'), 'N')
+  return sorted[#sorted]
+end
+
+configs[server_name] = {
+  default_config = {
+    cmd = { 'java' },
+    filetypes = { 'vdmsl', 'vdmpp', 'vdmrt' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.fn.getcwd()
+    end,
+    options = {
+      java = (vim.env.JAVA_HOME or '/usr') .. '/bin/java',
+      java_opts = { '-Xmx3000m', '-Xss1m' },
+      annotation_paths = {},
+      mavenrepo = mavenrepo,
+      version = get_latest_installed_version(mavenrepo),
+      logfile = util.path.join(vim.fn.stdpath 'cache', 'vdm-lsp.log'),
+      debugger_port = -1,
+      high_precision = false,
+    },
+  },
+  docs = {
+    description = [[
+https://github.com/nickbattle/vdmj
+
+The VDMJ language server can be installed by cloning the VDMJ repository and
+running `mvn clean install`.
+
+Various options are provided to configure the language server (see below). In
+particular:
+- `annotation_paths` is a list of folders and/or jar file paths for annotations
+that should be used with the language server;
+- any value of `debugger_port` less than zero will disable the debugger; note
+that if a non-zero value is used, only one instance of the server can be active
+at a time.
+
+More settings for VDMJ can be changed in a file called `vdmj.properties` under
+`root_dir/.vscode`. For a description of the available settings, see
+[Section 7 of the VDMJ User Guide](https://raw.githubusercontent.com/nickbattle/vdmj/master/vdmj/documentation/UserGuide.pdf).
+
+Note: proof obligations and combinatorial testing are not currently supported
+by neovim.
+]],
+    default_config = {
```

Comment:
You don't need defualt_config under documentation

Reply:
I think I do, since a couple of options aren't really meaningful without a custom description:
- https://github.com/neovim/nvim-lspconfig/blob/72e5b3f3952fa29d45a28a12257b6a6faf8dfc8f/lua/lspconfig/vdmj.lua#L47 will not have a meaningful value unless VDMJ is installed on the system
- https://github.com/neovim/nvim-lspconfig/blob/72e5b3f3952fa29d45a28a12257b6a6faf8dfc8f/lua/lspconfig/vdmj.lua#L37 is not representative of the actual default command generated in `on_new_config`

Reply:
Just put that under documentation then, as it stands you are repeating options that are auto-populated.

Reply:
If I leave only those two descriptions in the `default_config` table under documentation, then no other values are auto-populated and the other options do not appear in `CONFIG.md`, which is why I manually copied them over. I tried looking through `docgen.lua`, but I don't quite understand it and I can't figure out if there is some other way to direct it to generate the documentation for the other options.